staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -257,7 +270,15 @@ func (o *DebugOptions) Complete(restClientGetter genericclioptions.RESTClientGet
 	}
 
 	if o.Applier == nil {
-		applier, err := NewProfileApplier(o.Profile)
+		kflags := keepFlags{
```

Comment:
Agree that `keepFlags` is better naming and sorry, if I wasn't clear enough: Could you please export this, it's functions and it's fields?.

Reply:
> Could you please export this, it's functions and it's fields?.

It's ok.
Could you tell me the reason?
As you said [here](https://github.com/kubernetes/kubernetes/pull/123149#discussion_r1602072815), they are not used by anywhere so I thought we need not to export...

Reply:
> > Could you please export this, it's functions and it's fields?.
> 
> It's ok. Could you tell me the reason? As you said [here](https://github.com/kubernetes/kubernetes/pull/123149#discussion_r1602072815), they are not used by anywhere so I thought we need not to export...

Basically, in case some other tools depend on kubectl debug functionality, they would be able to customize the behavior. Otherwise, keep flag functionality will be black box. We can also may want to intentionally unexport them. However, they would fully be relying on the flags and it may better to export them in my opinion. Hope that it makes sense.

Reply:
Another reason would be `ProfileApplier` is exported function and can be customizable. On the other hand, it only gets an unexported type which drastically hurts the usability of `ProfileApplier`.

Reply:
Thank you for your explanation.
I've understood.